cluster-autoscaler/cloudprovider/aws/aws_cloud_provider.go

```diff
@@ -128,8 +128,18 @@ func (aws *awsCloudProvider) NodeGroupForNode(node *apiv1.Node) (cloudprovider.N
 }
 
 // HasInstance returns whether a given node has a corresponding instance in this cloud provider
-func (aws *awsCloudProvider) HasInstance(*apiv1.Node) (bool, error) {
-	return true, cloudprovider.ErrNotImplemented
+func (aws *awsCloudProvider) HasInstance(node *apiv1.Node) (bool, error) {
+	awsRef, err := AwsRefFromProviderId(node.Spec.ProviderID)
+	if err != nil {
+		return false, err
+	}
+
+	// we don't care about the instance health
+	if _, ok := aws.awsManager.asgCache.instanceStatus[*awsRef]; !ok {
```

Comment:
We already have `instanceStatus` field on `asgCache` to track health status by EC2 instance. I am re-using the `instantStatus` map to check if the node is present without caring for its health since we just want to check if the instance is present in the cloud provider. 

`instanceStatus` is updated at https://github.com/kubernetes/autoscaler/blob/1d56b81b578983b27dc8024defe9561605bd3273/cluster-autoscaler/cloudprovider/aws/auto_scaling_groups.go#L401-L404